spec/mediaelement-shared.js

```diff
@@ -32,6 +32,14 @@ export function sharedTests(backend) {
         loadElement();
     });
 
+    /**
+     * @test {WaveSurfer#waveformReady}
+     */
+    it('should be waveform ready', function(done) {
```

Comment:
I think this test should check that `waveform-ready` is only emitted once (not 2 or 3 times). This catches bugs where someone changes the if statements inside `loadMediaElement`. We can do this with jasmine spy functions:

```js
const waveformReadySpy = jasmine.createSpy('waveform-ready-spy');
waveform.once('waveform-ready', () => {
    waveformReadySpy();
});

// catch bugs where waveform-ready is called more than once during loading
expect(waveformReadySpy).toHaveBeenCalledTimes(1);
```

source:
https://stackoverflow.com/questions/34938016/testing-if-an-event-has-been-triggered-in-jasmine

Note: I don't use jasmine so maybe there is a better way to do this :)

Reply:
yeaah, thanks!